app/js/lib/foxbox/network.js

```diff
@@ -177,29 +179,30 @@ export default class Network {
    * @private
    */
   [p.pingBox]() {
-    const promises = [];
-    promises.push(this[p.ping](`${this.localOrigin}/ping`)
-      .then(() => {
-        this[p.local] = true;
-      })
-      .catch(() => {
-        this[p.local] = false;
-      }));
+    const previousState = this[p.local] || this[p.remote];
+
+    const promises = [
+      this[p.ping](`${this.localOrigin}/ping`).then(() => true, () => false),
+    ];
 
     // @todo Find a better way to detect if a tunnel connection is active.
     if (this[p.settings].tunnelOrigin) {
-      promises.push(this[p.ping](`${this.tunnelOrigin}/ping`)
-        .then(() => {
-          this[p.remote] = true;
-        })
-        .catch(() => {
-          this[p.remote] = false;
-        }));
+      promises.push(
+        this[p.ping](`${this.tunnelOrigin}/ping`).then(() => true, () => false)
+      );
     }
 
-    Promise.all(promises).then(() => {
-      this[p.pingCallback](this[p.local] || this[p.remote]);
-    });
+    Promise.all(promises)
+      .then(([isLocalOnline, isRemoteOnline]) => {
+        const currentState = isLocalOnline || isRemoteOnline;
+
+        this[p.local] = isLocalOnline;
+        this[p.remote] = isRemoteOnline;
+
+        if (currentState !== previousState) {
+          this.emit('online', currentState);
+        }
+      });
```

Comment:
What if one promise hangs on for a long time? It will delay the other one. I still think it's better not to use `Promise.all()` here.